Backend/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Data;
+using Data.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace Backend.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CategoryController : ControllerBase
+    {
+        private readonly Db _context;
+
+        public CategoryController(Db context)
+        {
+            _context = context;
+        }
+
+        // POST api/categories
+        [HttpPost]
+        public IActionResult CreateOrUpdateCategory([FromBody] string categoryName)
+        {
+            if (string.IsNullOrEmpty(categoryName))
+            {
+                return BadRequest("Category name cannot be empty.");
+            }
+
+            string lowercaseCategoryName = categoryName.ToLower();
+
+            var existingCategory = _context.Categories.FirstOrDefault(c => c.Name.ToLower() == lowercaseCategoryName);
+
+            if (existingCategory != null)
+            {
+                return Ok(existingCategory);
+            }
+            else
+            {
+                var newCategory = new Category(categoryName);
+                _context.Categories.Add(newCategory);
+                _context.SaveChanges();
+                return Ok(newCategory);
+            }
+        }
+
+        // DELETE api/categories/{categoryId}
+        [HttpDelete("{categoryId}")]
+        public IActionResult DeleteCategory(string categoryId)
```

Comment:
Should be async. And we don't have straight delete of category. It's dependent of shop. Like: who the f*ck delete category?